scipy/conftest.py

```diff
@@ -153,6 +155,33 @@ def check_fpu_mode(request):
 )
 
 
+# Following the approach of NumPy's conftest.py...
+# Use a known and persistent tmpdir for hypothesis' caches, which
+# can be automatically cleared by the OS or user.
+hypothesis.configuration.set_hypothesis_home_dir(
+    os.path.join(tempfile.gettempdir(), ".hypothesis")
+)
+
+# We register two custom profiles for SciPy - for details see
+# https://hypothesis.readthedocs.io/en/latest/settings.html
+# The first is designed for our own CI runs; the latter also
+# forces determinism and is designed for use via scipy.test()
+hypothesis.settings.register_profile(
+    name="nondeterministic", deadline=None, print_blob=True,
+)
+hypothesis.settings.register_profile(
+    name="deterministic",
+    deadline=None, print_blob=True, database=None, derandomize=True,
+    suppress_health_check=list(hypothesis.HealthCheck),
+)
```

Comment:
It looks like the timing-related flake was from a nondeterministically-exceeded [`deadline`, which is spiritually a healthcheck but uses a separate setting](https://hypothesis.readthedocs.io/en/latest/settings.html#hypothesis.settings.deadline).  Since you're already attending to test runtime, I'd suggest using `deadline=None` to disable it entirely.